src/datasources.js

```diff
@@ -11,14 +11,85 @@ define([
     edit_source,
     select_source
 ) {
-    var vellum, dataSources;
+    var vellum, dataSources, cachedDataSources;
 
     function init(instance) {
         vellum = instance;
         dataSources = vellum.opts().core.dataSources || [];
+        cachedDataSources = {
+            fixture: {}
+        };
+    }
+
+    function cacheFixtures(data) {
+        _.map(data, function(fixture) {
+            cachedDataSources.fixture[fixture.sourceUri] = fixture;
+        });
+    }
+
+    function getPossibleFixtures() {
+        function generateFixtureDefinitions(structure, baseFixture) {
+            return _.map(structure, function(value, key) {
+                var ret = [],
+                    newBaseFixture = {
+                        src: baseFixture.src,
+                        id: baseFixture.id,
+                        query: baseFixture.query + "/" + key
+                    };
+                newBaseFixture.name = baseFixture.name + " - " + (value.name || key);
+
+                if (!value.no_option) {
+                    ret = [newBaseFixture];
+                }
+                return ret.concat(generateFixtureDefinitions(value.structure, newBaseFixture));
+            });
+        }
+
+        return _.flatten(_.map(cachedDataSources.fixture, function(fixture) {
+            var baseFixture = {
+                src: fixture.sourceUri,
+                id: fixture.defaultId,
+                query: fixture.initialQuery,
+                name: fixture.name || fixture.defaultId
+            };
+
+            return [baseFixture].concat(generateFixtureDefinitions(fixture.structure, baseFixture));
+        }));
+    }
+
+    function generateFixtureOptions() {
+        return _.map(getPossibleFixtures(), function(fixture) {
+            return {
+                value: JSON.stringify(_.omit(fixture, 'name')),
+                text: fixture.name
+            };
+        });
+    }
+
+    function generateFixtureColumns(fixture) {
+        function generateColumns(structure) {
+            return _.map(structure, function(value, key) {
+                return [key].concat(_.map(generateColumns(value.structure), function(value) {
+                    return key + '/' + value;
+                }));
+            });
+        }
+
+        if (fixture) {
+            return _.flatten(generateColumns(fixture.structure));
+        }
+        return "";
```

Comment:
Would it make sense to return an empty array here rather than a string since the normal case is to return an array?